### PR TITLE
CA-350411: Add conditional page control refresh in Import Wizard

### PR DESCRIPTION
**Pull request by danilo-delbusso**
Page controls are now forcefully refreshed only when necessary.

I suggest removing whitespace diffs.

### CODE REVIEW

**Review by CitrixChris — DISMISSED** [review]

File: XenAdmin/Wizards/ImportWizard/ImportWizard.cs [hunk 1]
```diff
@@ -251,17 +251,30 @@ protected override void UpdateWizardContent(XenTabPage page)
                     m_pageImportSource.DiskCapacity, m_pageImportSource.IsWIM, m_pageVMconfig.AdditionalSpace,
                     m_pageImportSource.FilePath, m_pageImportSource.ImageLength);
 
+				if (m_pageStorage.SelectedOvfEnvelope != m_envelopeFromVhd
```
Comment (kc284) [inline XenAdmin/Wizards/ImportWizard/ImportWizard.cs:254]: An easier way to do this is assign the newly created envelope to a local variable and compare this with `m_envelopeFromVhd` before assigning to the latter:
```
var newEnv = OVF.Create(...);
if (newEnv != m_envelopeFromVhd) {...}
m_envelopeFromVhd = newEnv;
```

File: XenAdmin/Wizards/ImportWizard/ImportWizard.cs [hunk 2]
```diff
@@ -251,17 +251,30 @@ protected override void UpdateWizardContent(XenTabPage page)
...
-				NotifyNextPagesOfChange(m_pageHost, m_pageStorage, m_pageNetwork);
 			}
 			else if (type == typeof(ImportSelectHostPage))
 			{
 			    TargetConnection = m_pageHost.ChosenItem == null ? null : m_pageHost.ChosenItem.Connection;
                 RemovePage(m_pageRbac);
                 ConfigureRbacPage(TargetConnection);
+
+				if (m_pageNetwork.Connection != TargetConnection
+					|| m_pageStorage.VmMappings != m_vmMappings
```
Comment (kc284) [inline XenAdmin/Wizards/ImportWizard/ImportWizard.cs:273]: The m_vmMappings are set further down; at this point they have not changed yet, so the comparison with m_pageStorage.VmMappings is always true. However, I suggest using local variables for comparing the connection and mappings (like with the OvfEnvelope in the previous comment).

Also,  the removal/addition of the boot options page is not needed if the import type or connection have not changed.

**Review by kc284 — CHANGES_REQUESTED** [review]
We can also add checks for the page types ImportSourcePage and GlobalSelectHost.

**Review by AaronRobson — DISMISSED** [review]

**Comment by kc284** [comment]
This one proved trickier than anticipated and there was also an additional bug whereby the Boot options selected by the user were not set correctly when one imported VM from image. I have submitted some corrections here https://github.com/danilo-delbusso/xenadmin/pull/9. The whole thing will need re-review once merged and a lot of testing because there is a risk of the changes introducing regressions.

**Review by kc284 — CHANGES_REQUESTED** [review]
Setting it status to "changes requested" until we review and merge https://github.com/danilo-delbusso/xenadmin/pull/9

File: XenAdmin/Wizards/ImportWizard/NetworkPickerPage.cs [hunk 3]
```diff
@@ -49,8 +49,8 @@ public partial class NetworkPickerPage : XenTabPage
 		#region Private fields
 		private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
 		
-		private IXenConnection m_selectedConnection;
-        private Host m_selectedAffinity;
+		internal IXenConnection MSelectedConnection { get; set; }
```
[the remark this answers is not in the document]
Comment (danilo-delbusso) [reply XenAdmin/Wizards/ImportWizard/NetworkPickerPage.cs:52]: Thanks! I was wondering what that was 😂

**Review by AaronRobson — APPROVED** [review]

**Review by CitrixChris — APPROVED** [review]

**Pull request merged**